s3tests_boto3/functional/test_sts.py

```diff
@@ -249,6 +251,7 @@ def test_assume_role_allow():
     eq(s3bucket['ResponseMetadata']['HTTPStatusCode'],200)
     bkt = s3_client.delete_bucket(Bucket=bucket_name)
     eq(bkt['ResponseMetadata']['HTTPStatusCode'],204)
+    iam_client.delete_role_policy(RoleName=general_role_name,PolicyName=policy_name)
```

Comment:
removing user policy for get session token makes sense. But I am not sure why deleting the role policy is needed for assume role tests, since the code clean up makes use of permanent credentials and role policy only affects temporary credentials (unless ofcourse there is a hidden bug). This needs to be debugged a bit more locally.

Reply:
Yes @pritha-srivastava you're correct. I guess we were seeing the error due to get_session_token tests (as per what I've checked).